Homework5/SparseMatrix/Entities/SparseMatrix.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SparseMatrixOperations.Entities
+{
+    public class SparseMatrix : IEnumerable<int>
+    {
+        private Dictionary<(int Column, int Row), int> MatrixElements = new Dictionary<(int, int), int>();
+        private (int Columns, int Rows) Size;
+
+
+        public SparseMatrix(int columns, int rows)
+        {
+            if (columns > 0 && rows > 0)
+            {
+                Size = (columns, rows);
+            }
+            else
+            {
+                throw new ArgumentException("Size of matrix isn't appropriate.");
+            }
+        }
+
+        public int this[int col, int row]
+        {
+            get
+            {
+                InputDimensionCheck(col, row);
+
+                if (IsZeroElement(col, row)) // case when there is no element in such indexes, so the value is zero
+                {
+                    return default(int);
+                }
+
+                var elementIndex = (col, row);
+                return MatrixElements.First(e => e.Key.Equals(elementIndex)).Value; // returning the value at corresponding indexes
```

Comment:
access an element via Dictonary getter, do not call .First, it iterates over the all element under the hood.